### PR TITLE
Add support for getting product tags by collection

### PR DESCRIPTION
**Pull request by gabrieloc**
## What this does
Adds a new method to `BUYClient+Storefront` allowing a collection ID to be passed in for getting tags, effectively allowing a subset of available tags to be fetched.

@bgulanowski @davidmuzi @krisorr

### CODE REVIEW

File: Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m [hunk 1]
```diff
@@ -126,10 +126,20 @@ - (NSOperation *)getProductsByIds:(NSArray *)productIds completion:(BUYDataProdu
...
+	return [self getProductTagsInCollection:@"" page:page completion:block];
+}
+
+- (NSOperation *)getProductTagsInCollection:(NSString *)collectionId page:(NSUInteger)page completion:(BUYDataTagsListBlock)block
+{
+	NSMutableDictionary *params = @{
+									@"limit" : @(self.pageSize),
+									@"page"  : @(page),
+									}.mutableCopy;
+	if (collectionId.length) {
```
Comment (krisorr) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: we made collectionId mandatory on Android, to match the other APIs

**Comment by bgulanowski** [comment]
Agree with Kris. Otherwise LGTM.

File: Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m (hunk 1 above)
Comment (gabrieloc) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: `collectionId` is only privately optional, as this line is/should only be exercised by line 129 ( @bgulanowski )

Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: Oh I see, we are relying on nullability? That works.

**Comment by gabrieloc** [comment]
Test failures unrelated

**Pull request merged**